File: src/main/java/net/glasslauncher/mods/glasstech/GTWireMaterial.java
package net.glasslauncher.mods.glasstech;

public class GTWireMaterial {
    public static final GTWireMaterial IRON = new GTWireMaterial("iron_cable", -1, VoltageTier.LV.maxVoltage * 8, VoltageTier.LV, 4);
    public static final GTWireMaterial ALUMINIUM = new GTWireMaterial("aluminium_cable", -1, VoltageTier.MV.maxVoltage * 10, VoltageTier.LV, 3);
    public static final GTWireMaterial CARBON = new GTWireMaterial("carbon_cable", -1, VoltageTier.MV.maxVoltage * 16, VoltageTier.LV, 2);
    public static final GTWireMaterial GOLD = new GTWireMaterial("gold_cable", -1, VoltageTier.HV.maxVoltage * 7, VoltageTier.LV, 2);
    public static final GTWireMaterial COPPER = new GTWireMaterial("copper_cable", -1, VoltageTier.HV.maxVoltage * 8, VoltageTier.LV, 2);
    public static final GTWireMaterial SILVERED_COPPER = new GTWireMaterial("silvered_copper_cable", -1, VoltageTier.HV.maxVoltage * 8, VoltageTier.LV, 1);
    public static final GTWireMaterial SILVER = new GTWireMaterial("silver_cable", -1, VoltageTier.HV.maxVoltage * 12, VoltageTier.LV, 1);

    public final String translationKey;
    public final int color;
    public final int maxVoltage;
    public final VoltageTier voltageTier;
    public final int lossPerBlock;


    private GTWireMaterial(String translationKey, int color, int maxVoltage, VoltageTier voltageTier, int lossPerBlock) {
        this.translationKey = translationKey;
        this.color = color;
        this.maxVoltage = maxVoltage;
        this.voltageTier = voltageTier;
        this.lossPerBlock = lossPerBlock;
    }
}

File: src/main/kotlin/net/glasslauncher/mods/glasstech/GTWireMaterial.kt
package net.glasslauncher.mods.glasstech

import net.teamterminus.machineessentials.energy.electric.api.VoltageTier
import net.teamterminus.machineessentials.energy.electric.api.WireMaterial

class GTWireMaterial {
    companion object {
        val IRON = WireMaterial("iron_cable", -1, VoltageTier.LV.maxVoltage * 8, VoltageTier.LV, 4, 1000)
        val ALUMINIUM = WireMaterial("aluminium_cable", -1, VoltageTier.MV.maxVoltage * 10, VoltageTier.LV, 3, 1000)
        val CARBON = WireMaterial("carbon_cable", -1, VoltageTier.MV.maxVoltage * 16, VoltageTier.LV, 2, 1000)
        val GOLD = WireMaterial("gold_cable", -1, VoltageTier.HV.maxVoltage * 7, VoltageTier.LV, 2, 1000)
        val COPPER = WireMaterial("copper_cable", -1, VoltageTier.HV.maxVoltage * 8, VoltageTier.LV, 2, 1000)
        val SILVERED_COPPER = WireMaterial("silvered_copper_cable", -1, VoltageTier.HV.maxVoltage * 8, VoltageTier.LV, 1, 1000)
        val SILVER = WireMaterial("silver_cable", -1, VoltageTier.HV.maxVoltage * 12, VoltageTier.LV, 1, 1000)
    }
}